tensorly/tenalg/svd.py

```diff
@@ -357,6 +358,7 @@ def randomized_svd(
 
 
 SVD_FUNS = ["truncated_svd", "symeig_svd", "randomized_svd"]
+SVD_TYPES = Literal["truncated_svd", "symeig_svd", "randomized_svd"]
```

Comment:
Should it just be Literal[f for f in SVD_FUNS] ?

Reply:
Unfortunately, `Literal` [doesn't allow arbitrary values](https://typing.readthedocs.io/en/latest/spec/literal.html#illegal-parameters-for-literal-at-type-check-time). A hacky workaround is

```
SVD_TYPES = Literal["truncated_svd", "symeig_svd", "randomized_svd"]
SVD_FUNS = list(SVD_TYPES.__args__)
```

but I'm not sure if that's any better than just duplicating the string, which makes it absolutely clear for the reader what's happening with the code.